### ISSUE TITLE
Version 0.8.2.1 is not pep440 compliant

### ISSUE THREAD

**Issue by BradHolmes** [ilanschnell/bitarray#58]
Hello,

Following on the recent problems with 0.8.2, it appears that the latest version is uploaded to pypi as 0.8.2.1, despite the fact that it says at https://pypi.org/project/bitarray/0.8.2/#history it is 0.8.2.

```
pip install bitarray==0.8.2
```

does not work.  It reports:

```
Could not find a version that satisfies the requirement bitarray==0.8.2 (from -r foo (line 1)) (from versions: 0.1.0, 0.2.0, 0.2.1, 0.2.2, 0.2.3, 0.2.4, 0.2.5, 0.3.0, 0.3.1, 0.3.2, 0.3.3, 0.3.4, 0.3.5, 0.4.0, 0.5.0, 0.5.1, 0.5.2, 0.6.0, 0.7.0, 0.8.0, 0.8.1, 0.8.2.1)

No matching distribution found for bitarray==0.8.2 (from -r foo (line 1))
```

Is it possible to deploy this as version 0.8.3, and remove 0.8.2.1 completely from pypi?  Or alternatively, fix the 0.8.2 deployment to be saved as 0.8.2?

**Comment by ilanschnell** [comment]
The version is 0.8.2, but the tarball is labeled 0.8.2.1 because I had to fix something, and PyPI will not allow reuploading a tarball with the same filename.

**Comment by BradHolmes** [comment]
Correct, I started with #57 because we were burned by that as well.  I appreciate the quick response on that one.

So, I think there are two problems, and I am unsure which one contributes to `pip freeze` reporting `0.8.2`.  When it does, that makes pip freeze/install cycles fail:

```
pip freeze > requirements.txt
pip install -r requirements.txt
```

pip freeze reports 0.8.2, so the second command attempts `pip install bitarray==0.8.2`.  But that does not exist in PyPI.  

It is either due to:
* the tarball does not match the self-reporting version
* the version string is not pep400 compliant (only 3 fields allowed AFAIU), and `pip freeze` is truncating the version string.

What if you tried a [post-release version identifier](https://www.python.org/dev/peps/pep-0440/#post-releases)?

**Comment by sekrause** [comment]
I think the best solution is to release a 0.8.3, then there is no confusion anymore.

**Comment by ilanschnell** [comment]
I've released 0.8.3 to avoid the confusion.
